pkg/generator/appconfiguration/app_configurations_generator.go

```diff
@@ -21,11 +21,12 @@ func (acg *Generator) GenerateSpec(
 		Resources: []models.Resource{},
 	}
 
-	g, err := generators.NewAppConfigurationGenerator(project.Name, acg.AppConfiguration)
-	if err != nil {
-		return nil, err
-	}
-	if err = g.Generate(spec); err != nil {
+	gfs := []generators.NewGeneratorFunc{}
+	generators.ForeachOrderedApps(acg.Apps, func(appName string, app appmodel.AppConfiguration) error {
```

Comment:
It's a little over-designed that we add a function into the params. Considering that we only invoked this function once, we can inline the `func(appName string, app appmodel.AppConfiguration) error` in the implication of `ForeachOrderedApps`

Reply:
Next PR, I will optimize this method using `Generics`.